Mathlib/LinearAlgebra/Matrix/Determinant/Basic.lean

```diff
@@ -183,14 +183,14 @@ theorem det_mul_left_comm (M N P : Matrix m m R) : det (M * (N * P)) = det (N *
 theorem det_mul_right_comm (M N P : Matrix m m R) : det (M * N * P) = det (M * P * N) := by
   rw [Matrix.mul_assoc, Matrix.mul_assoc, det_mul, det_mul_comm N P, ← det_mul]
 
--- TODO(mathlib4#6607): fix elaboration so that the ascription isn't needed
+-- TODO(mathlib4#6607): fix elaboration so `val` isn't needed
 theorem det_units_conj (M : (Matrix m m R)ˣ) (N : Matrix m m R) :
-    det ((M : Matrix _ _ _) * N * (↑M⁻¹ : Matrix _ _ _)) = det N := by
+    det (M.val * N * M⁻¹.val) = det N := by
```

Comment:
Does
```suggestion
    det (M * N * M⁻¹ : Matrix m m _) = det N := by
```
work here? Up do you if you think that's better.

Reply:
No, it does not work:
```
failed to synthesize
  HMul (Matrix m m R)ˣ (Matrix m m R) ?m.90214
 ```
 for the first multiplication `M * N`